src/web/app/components/question-types/question-constraint/rank-recipients-question-constraint.component.ts

```diff
@@ -20,17 +22,20 @@ import {
   templateUrl: './rank-recipients-question-constraint.component.html',
   styleUrls: ['./rank-recipients-question-constraint.component.scss'],
 })
-export class RankRecipientsQuestionConstraintComponent implements OnInit {
+export class RankRecipientsQuestionConstraintComponent extends QuestionConstraintComponent implements OnInit {
 
   @Input()
   questionDetails: FeedbackRankRecipientsQuestionDetails = DEFAULT_RANK_RECIPIENTS_QUESTION_DETAILS();
 
   @Input()
   recipientSubmissionForms: FeedbackResponseRecipientSubmissionFormModel[] = [];
```

Comment:
Try to use setter method here.

Reply:
What do you mean by this? To set the form model as an alternative to `@Input`?

Reply:
And thus, if you use `setter` method (https://angular.io/guide/component-interaction#intercept-input-property-changes-with-a-setter) here:

This is where you should fire the event.